core/src/Revolution/Processors/Element/Create.php

```diff
@@ -50,14 +58,47 @@ public function cleanup()
      */
     public function beforeSave()
     {
-        $nameField = $this->classKey === modTemplate::class ? 'templatename' : 'name';
-        $name = $this->getProperty($nameField, '');
-
-        /* verify element with that name does not already exist */
-        if ($this->alreadyExists($name)) {
-            $this->addFieldError($nameField, $this->modx->lexicon($this->objectType . '_err_ae', [
-                'name' => $name,
-            ]));
+        $locked = (bool)$this->getProperty('locked', false);
+        $this->object->set('locked', $locked);
+
+        $elementClassName = array_pop(explode('\\', $this->classKey));
+
+        if ($elementClassName === 'modTemplateVar') {
```

Comment:
Is there a reason why you're not using the fully qualified classname here because doing it like this would make it harder to maintain down the road.

Reply:
Simply that I was not as wise then (when I originally submitted this) as I am today. Changed to FQC here and in the Update class (commit [c6812e4](https://github.com/modxcms/revolution/pull/15936/commits/c6812e424c946107895377f17cfbcf4f0b52c0db).)